app/components/inventory-number-input.js

```diff
@@ -24,7 +24,23 @@ export default Ember.Component.extend({
       };
       var onError = error => this.get("messageBox").alert("Scanning failed: " + error);
       var options = {"formats": "CODE_128"};
-      cordova.plugins.barcodeScanner.scan(onSuccess, onError, options);
+      // add camera permission for scanning barcode here
```

Comment:
**Few comments :** 

- The `permissionError` only logs text in the console, which would result in nothing happening. Probably better to use `onError('Camera permission is not turned on')` so that the user knows what's happening

- On line 39 `if( !status.hasPermission ) { permissionError(); }` ,
an error is logged, but isn't followed by any return statement (could be an if/else as well). That means that the next line will execute, will try to scan the code, and break. 
We basically need to make sure`barcodeScanner.scan` isn't run unless we're sure we have permission.

- The following block of code
```javascript
var success = (status) => {
  if( !status.hasPermission ) { permissionError(); }
    cordova.plugins.barcodeScanner.scan(onSuccess, onError, options);
};
```
is currently written _after_ being used. That works because of how the javascript compiler manages variable declarations (it actually secretly declares the `var success` at the top of the block, so the reference already exists), but it's dangerous to presume so. Better to have it declared before :)
In that regard, it's also better to use `let` over `var`, and `const` whenever a variable shouldn't be modified (not a big deal though)

**Some tips (just for fun, use them only if you like and find them useful)** 

- The `status` variable is only used to access it's own `hasPermission` property. You could use ES6 destructuring like this :
```javascript
function({ hasPermission }) {
  // use hasPermission directly
}
```

Note: We seem have a lot of strings in components which are not in the translation files, we should work on slowly moving them there, and using the i18n service

Reply:
@patrixr, thanks for the suggestions, loved them, will apply to my changes, and let you know.